### ISSUE TITLE
Problem of parameters with Wifi&FMCW

### ISSUE THREAD

**Issue by whitecloudy** [mobicom24/RF-Diffusion#17]
Hello,

I have recently attempted to perform inference for Wi-Fi and FMCW using the provided code. However, I have noticed an issue with the parameters used in these experiments.

According to the paper, the value of self.alpha_bar[T], corresponding to the gamma_hat(T) in the T-step, should be converges to 0. However, the noise_schedule for both Wi-Fi and FMCW seems to be set to values that are too small, causing self.alpha_bar[T] to become significantly larger than it should be.

As a result, when running inference with "fast_sampling"(which starts from gaussian noise) instead of "native_sampling", the results of the paper cannot be reproduced. This outcome is fundamentally incorrect, based on both the basic principles of Diffusion models and the claims of your paper.

Could you please review this issue and provide clarification? I look forward to your response.

Thank you!
@Guoxuan-Chi

**Comment by whitecloudy** [comment]
@mobicom24

**Comment by jiangtguo** [comment]
May I ask how you debugged it? My inference result shows that the FID value is around 140 and the Average SSIM is around 0.7. This result is terrible

**Comment by whitecloudy** [comment]
@jiangtguo Still has same issue. I also couldn't reproduce the same FID or SSIM result with Widar dataset.(Not Widar performance perspective. I haven't tried it yet.) Since we don't have training dataset for Wi-fi test, unless the author comment something about this issue, it seems deadend for now. I am trying to reproduce the MIMO result with Argos dataset for now.

**Comment by jiangtguo** [comment]
Thank you for your reply. If there is any method to obtain robust results, I hope you can share it with me

**Comment by sunday575** [comment]
> 你好
> 
> 我最近尝试使用提供的代码对 Wi-Fi 和 FMCW 执行推理。但是，我注意到这些实验中使用的参数存在问题。
> 
> 根据该论文，对应于 T 步长中 gamma_hat（T） 的 self.alpha_bar[T] 的值应收敛为 0。但是，Wi-Fi 和 FMCW 的 noise_schedule 似乎都设置为太小的值，导致 self.alpha_bar[T] 变得明显大于应有的大小。
> 
> 因此，当使用 “fast_sampling”（从高斯噪声开始）而不是 “native_sampling” 进行推理时，无法重现论文的结果。根据 Diffusion 模型的基本原理和您论文的主张，这个结果从根本上是不正确的。
> 
> 您能否回顾一下这个问题并提供澄清？我期待着您的回复。
> 
> 谢谢！

你好，关于这个数据的问题，是怎么从n*181维度的数据转换到n*90的维度的

**Comment by zhangcf0309** [comment]
> May I ask how you debugged it? My inference result shows that the FID value is around 140 and the Average SSIM is around 0.7. This result is terrible

My inference result shows that the FID value is around 4.56 and the Average SSIM is around 0.001.

**Comment by whitecloudy** [comment]
> > May I ask how you debugged it? My inference result shows that the FID value is around 140 and the Average SSIM is around 0.7. This result is terrible
> 
> My inference result shows that the FID value is around 4.56 and the Average SSIM is around 0.001.

It does not when you use "fast_sampling" with Wi-FI or FMCW. And also, you can see that "degrade_fn" in native_sampling uses fixed random seed for forward process.
https://github.com/mobicom24/RF-Diffusion/blob/eb872b0c4543da65424f5598ae40826e76e7edea/tfdiff/diffusion.py#L82
